### PR TITLE
Reflect that Tor version must be  >= 0.3.2.2-alpha to reach V3 onions.

### LINKED ISSUE #3642: Onion V3 addresses only work on recent tor versions.
Ubuntu 18.04 ships tor 0.2.9 by default, which doesn't recognise onion v3 addresses, which leads to some trouble with nodes that only advertise onion v3 addresses. There's quite a lot of them.

I think that:
 - the documentation on TOR should be updated, regarding the version.
 - lightningd should recognise nodes that only advertise .onion addresses and should warn the user that the connection error was caused by a lack of tor proxy.
 - lightningd should check the TOR version and warn if it's too old.

This is how it looks when the tor version is too old:

```
$ lightning-cli connect 02d4b432058ec31e38f6f35d22a487b7db04c4bf70f201f601b66f7b4358242b03@2b3d5dc7ejbnaqmrsdx3stopvswyhx7l2rtrddgqwgwhjn436k5hx7qd.onion:9735
{
   "code": 401,
   "message": "2b3d5dc7ejbnaqmrsdx3stopvswyhx7l2rtrddgqwgwhjn436k5hx7qd.onion:9735: Connection establishment: Operation now in progress. 2b3d5dc7ejbnaqmrsdx3stopvswyhx7l2rtrddgqwgwhjn436k5hx7qd.onion:9735: Connection establishment: Operation now in progress. "
}

$ torify nc anhvjzifi7oshenj5mqxmscsv3epux6kcesm2q2we6qhig2pcz2cuead.onion 9735
[Apr 12 14:55:58] ERROR torsocks[6198]: General SOCKS server failure (in socks5_recv_connect_reply() at socks5.c:516)
```

### PR DESCRIPTION
**Pull request by Saibato**
- Reflect in TOR.md that we need a Tor version >= 0.3.2.2-alpha if we want to reach Tor V3 onions 
- change the error response  

fix #3642 reported by @thestick613

### CODE REVIEW

File: doc/TOR.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
 
+Please note that nodes with V3 onion address i.e `vww6ybal4bd7szmgncyruucpgfkqahzddi37ktceo3ah7ngmcopnpyyd.onion`
+will not be reachable over Tor if your Tor version is below 0.3.2.2-alpha
+
+Connections to nodes with old Tor V2 address form with less than 10 char prefix before .onion
+i.e.`3fyb44wdhnd2ghhl.onion` should work with any version of Tor.
+
+You can check your installed Tor version with `tor --version` or `sudo tor --version`
+
+if not installed you can install Tor on Linux by
```
Comment (cdecker) [inline doc/TOR.md:13]:
```suggestion
If Tor is not installed you can install it on Debian based Linux systems (Ubuntu, Debian, etc) with the following command:
```

Comment (Saibato) [reply doc/TOR.md]: done :+1:

File: connectd/connectd.c [hunk 2]
```diff
@@ -649,6 +649,11 @@ static void destroy_io_conn(struct io_conn *conn, struct connecting *connect)
 		errstr = "peer closed connection";
 		if (streq(connect->connstate, "Cryptographic handshake"))
 			errstr = "peer closed connection (wrong key?)";
+	} else {
+		if (connect->addrs[connect->addrnum].u.wireaddr.type == ADDR_TYPE_TOR_V3) {
+			errstr = tal_fmt(tmpctx, "%s. Hint: check debuglog and can your tor version handle a V3 onion address?",
```
Comment (Saibato) [inline connectd/connectd.c:654]: A bit ugly to do it here\idk/
Why not pass 'connect->errors' to tor.c and add the error string from debug  there  
:woman_shrugging: ???

Comment (Saibato) [reply connectd/connectd.c]: hmm, reviewed  my own code, seam to me my quantum Grover time body split worked.
Anyway I like this, so i will change the code and add an error_add function.
thx to my other self for review. :+1:

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3643/commits/5580ce2c69fc9072376aa916d0a7f397ca58ab2d

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3643/commits/b6445267272218bf5574424bb0e83160aff629c4

**Pull request merged**